state/action.go

```diff
@@ -449,20 +458,23 @@ func (a *action) removeAndLogBuildTxn(finalStatus ActionStatus, results map[stri
 					}
 				}
 				updateOperationOp = &txn.Op{
-					C:      operationsC,
-					Id:     a.st.docID(parentOperation.Id()),
-					Assert: assertNotComplete,
+					C:  operationsC,
+					Id: a.st.docID(parentOperation.Id()),
+					Assert: append(assertNotComplete,
+						bson.DocElem{"complete-task-count", bson.D{{"$eq", expectedTaskCount - 1}}}),
```

Comment:
```suggestion
						bson.DocElem{"spawned-task-count", bson.D{{"$eq", expectedTaskCount - 1}}}),
```

Reply:
when completing an operation, the complete task count should always be one less than the spawned count.